pkg/cmd/server/api/types.go

```diff
@@ -0,0 +1,143 @@
+	NodeConfig            NodeConfig
+}
+
+type NodeConfig struct {
+	api.TypeMeta
+
+	NodeID string
+
+	NodeHost string
+
+	KubernetesAddress string
+	KubernetesCA      FileOrData
+
+	// Whether to enable TLS serving
+	TLS bool
+	// ServingCert is used to serve securely
+	ServingCert CertInfo
+
+	// The directory that volumes will be stored under
+	VolumeDir string
+
+	// The image used as the Kubelet network namespace and volume container.
+	NetworkContainerImage string
+
+	// If true, the Kubelet will ignore errors from Docker
+	AllowDisabledDocker bool
+
+	// KubernetesCredentials is the client configuration used identify this kubelet to the master
+	KubernetesCredentials KubernetesCredentials
+
+	RecordEvents bool
+}
+
+type OpenShiftMasterConfig struct {
+	api.TypeMeta
+
+	// host:port to bind master to
+	MasterBindAddress string
+	// url to access the master API on within the cluster
+	MasterAddress string
+	// external clients may need to access APIs at different addresses than internal components do
+	MasterPublicAddress string
+
+	// host:port to bind asset server to
+	AssetBindAddress   string
+	AssetPublicAddress string
+
+	// LogoutURI is an optional, absolute URI to redirect web browsers to after logging out of the web console.
+	// If not specified, the built-in logout page is shown.
+	LogoutURI string
+
+	EtcdURL    string
+	EtcdConfig *EtcdConfig
+
+	// url to access kubernetes API on within the cluster
+	KubernetesAddress       string
+	KubernetesPublicAddress string
+	KubernetesCA            FileOrData
+
+	// KubernetesMasterConfig is used to start a kubernetes master in process.  It is optional
+	KubernetesMasterConfig *KubernetesMasterConfig
+
+	CORSAllowedOrigins []string
+
+	RootCA string
+
+	// DeployerCert is the client configuration used to call OpenShift APIs from launched deployer pods
+	DeployerCert CertInfo
+	// OpenShiftSystemCert is the client configuration used to call OpenShift APIs from system components
+	OpenShiftSystemCert CertInfo
+	// KubernetesCredentials is the client configuration used to call Kubernetes APIs from system components.
+	KubernetesCredentials KubernetesCredentials
+
+	// MasterCerts are used to serve the apiserver securely
+	MasterCerts CertInfo
+	// AssetCerts are used to serve the assets securely
+	AssetCerts CertInfo
+
+	MasterAuthorizationNamespace string
+
+	ImageFormat     string
+	UseLatestImages bool
+}
+
+type FileOrData struct {
+	Filename string
+	Data     []byte
+}
+
+type KubernetesCredentials struct {
+	Username string
+	Password string
+	Token    string
+	Cert     FileOrData
+	Key      FileOrData
+}
+
+type EtcdConfig struct {
+	BindAddress   string
+	PeerAddress   string
+	MasterAddress string
+	StorageDir    string
+}
+
+type KubernetesMasterConfig struct {
+	api.TypeMeta
+
+	MasterHost string
+	MasterPort int
+
+	EtcdURL string
+
+	// KubernetesSystemCert is the client configuration used to call Kubernetes APIs from system components.
+	KubernetesSystemCert CertInfo
```

Comment:
No.  The kubernetest master config is only used when starting kube with us and we use certs for the connection.